### DISCUSSION TITLE
devtool doesn't show value of getter

### DISCUSSION THREAD

**Discussion by lake2** [pmndrs/valtio#733] [Q&A] [answered]
```ts
class State {
    spinning = false;

    get number() {
        return this.spinning ? 1 : 0;
    }

    toggle() {
        this.spinning = !this.spinning;
    }
}

const state_3 = proxy(new State());

const Count: React.FC = React.memo(function Count(props) {
    return (
        <div>
            <button onClick={() => state_3.toggle()}>start 3</button>
            <div>{snap3.number}</div>
        </div>
    );
});
```
state
![image](https://github.com/pmndrs/valtio/assets/6293752/f4f0e4ef-3ba4-4f69-9484-bc572b9a7e6e)

diff
![image](https://github.com/pmndrs/valtio/assets/6293752/80b4a001-c849-427d-8930-0361d0518915)

**Comment by dai-shi** [comment, marked as answer]
Thanks for reporting.

Class getters are not own properties, and it's not handled by `snapshot()`. It's by design.

An easy workaround is to use object getters.
```js
const state_3 = proxy({
  spinning: false,
  get number() {
    return this.spinning ? 1 : 0;
  },
  toggle() {
    this.spinning = !this.spinning;
  },
});
```
Or maybe coping prototype after instantiating a class...

There might be a devtool specific solution, but that would be fairly complicated and not very maintainable.

**Reply by lake2** [reply]
thanks
